packages/opencensus-exporter-jaeger/src/jaeger-driver/index.ts

```diff
@@ -148,3 +148,29 @@ export function spanToThrift(span: Span): ThriftSpan {
     logs: spanLogs,
   };
 }
+
+function getThriftReference(links: Link[]): ThriftReference[] {
+  return links
+             .map(link => {
+               const refType = getThriftType(link.type);
+               if (!refType) return null;
+
+               const traceId =
+                   `00000000000000000000000000000000${link.traceId}`.slice(-32);
+               const traceIdHigh = Utils.encodeInt64(traceId.slice(0, 16));
+               const traceIdLow = Utils.encodeInt64(traceId.slice(16));
+               const spanId = Utils.encodeInt64(link.spanId);
+               return {traceIdLow, traceIdHigh, spanId, refType};
+             })
+             .filter(ref => !!ref) as ThriftReference[];
```

Comment:
Would it work to move the `ThriftReference` type into the map function, i.e. `.map((link): ThriftReference|null => {`?

That would make the compiler be stricter in how it matches the type I believe.

Reply:
Done